frontend/TestRun/SCT/SctEvent.svelte

```diff
@@ -0,0 +1,194 @@
+<script lang="ts">
+    import humanizeDuration from "humanize-duration";
+    import { timestampToISODate } from "../../Common/DateUtils";
+    import type { EventSeverityFilter, Options, SCTEvent } from "./SctEvents.svelte";
+    import { faNode } from "@fortawesome/free-brands-svg-icons";
+    import Fa from "svelte-fa";
+    import { faClipboard, faClock, faServer, faSpider } from "@fortawesome/free-solid-svg-icons";
+
+    interface Props {
+        event: SCTEvent,
+        filterState: EventSeverityFilter,
+        options: Options,
+    }
+    let { event, filterState, options }: Props = $props();
+
+    const sliceMessage = function (message: string): string {
+        return fullMessage ? message : message.slice(0, Math.max(Math.ceil(message.length * 0.15), 500));
+    }
+
+    const messageRegex = /\((?<eventType>\w+) Severity\.(?<severity>[A-Z]+)\) (?<rawFields>.+)\n?/s;
+    const parseEventFields = function (rawFields: string) {
+        let pos = 0;
+        let currentChar;
+        let splitPoints = [pos];
+        let potentialSplit = false;
+        let potentialSplitPos = 0;
+        while (currentChar = rawFields.at(pos++)) { // eslint-disable-line
+            if (currentChar == " ") {
+                potentialSplit = true;
+                potentialSplitPos = pos;
+            } else if (potentialSplit && currentChar == "=") {
+                splitPoints.push(potentialSplitPos);
+                potentialSplit = false;
+                potentialSplitPos = -1;
+            }
+        }
+
+        let parsed = splitPoints.reduce((acc, value, idx, src) => {
+            let field = rawFields.slice(value, src[idx + 1]);
+            let [fieldKey, fieldVal] = field.split("=", 2);
+            if (!fieldKey || !fieldVal) return acc;
+            acc[fieldKey.trim()] = fieldVal.trim().replace(/[:,]$/, "");
+            return acc;
+        }, {} as ParsedFields);
+
+        return parsed;
+    };
+
+    const parseEventMessage = function (message: string) {
+        let eventMetaIndex = message.match(/\n/)?.index;
+        const eventMeta = message.slice(0, eventMetaIndex);
+        const logLine = message.slice(eventMetaIndex);
+        const match = messageRegex.exec(eventMeta);
+        if (!match) throw new Error("Unable to parse the event");
+        let parsedMessage = Object.assign({}, match.groups as unknown) as ParsedMessage;
+        let parsedFields = parseEventFields(parsedMessage.rawFields.trim());
+
+        parsedMessage.fields = parsedFields;
+        parsedMessage.logLine = logLine.trim();
+        parsedMessage.parsed = true;
+        parsedMessage.message = message;
+
+        return parsedMessage;
+    };
+
+    let fullMessage = $state(false);
+    //let parsedMessage = $derived(parseEventMessage(event.message));
```

Comment:
leftover code

Reply:
Not leftover - re-enabled in latest commit